api/app/src/test/kotlin/packit/integration/controllers/OutpackControllerTest.kt

```diff
@@ -0,0 +1,116 @@
+package packit.integration.controllers
+
+import org.junit.jupiter.api.Test
+import org.springframework.http.*
+import packit.integration.IntegrationTest
+import kotlin.test.Ignore
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+
+class OutpackControllerTest : IntegrationTest()
+{
+    val testPacket = "{\n" +
+            "  \"schema_version\": \"0.0.1\",\n" +
+            "  \"name\": \"modup-201707-queries1\",\n" +
+            "  \"id\": \"20170818-164847-7574883b\",\n" +
+            "  \"time\": {\n" +
+            "    \"start\": 1503074938.2232,\n" +
+            "    \"end\": 1503074938.2232\n" +
+            "  },\n" +
+            "  \"parameters\": null,\n" +
+            "  \"files\": [],\n" +
+            "  \"depends\": [],\n" +
+            "  \"script\": [\n" +
+            "    \"script.R\"\n" +
+            "  ],\n" +
+            "  \"session\": {\n" +
+            "  },\n" +
+            "  \"custom\": null\n" +
+            "}"
+
+    @Test
+    fun `can GET json from outpack_server`()
+    {
+        val result = restTemplate.getForEntity("/outpack/metadata/list", String::class.java)
+        assertSuccess(result)
+        jsonValidator.validateAgainstOutpackSchema(result.body!!, "location")
+    }
+
+    @Test
+    fun `can GET plain text from outpack_server`()
+    {
+        val result = restTemplate.getForEntity(
+                "/outpack/metadata/20230427-150755-2dbede93/text",
+                String::class.java
+        )
+        assertEquals(result.statusCode, HttpStatus.OK)
+        assert(result.headers.contentType.toString().contains("text/plain"))
+    }
+
+    @Test
+    fun `can GET file from outpack_server`()
+    {
+        val result = restTemplate.getForEntity(
+                "/outpack/file/sha256:1e2e932aa25493f54366fef8ec996a24ff3456c6b30d4ff6fa753e6263cf8ee0",
+                String::class.java
+        )
+        assertEquals(result.statusCode, HttpStatus.OK)
+        assertEquals(result.headers.contentType, MediaType.APPLICATION_OCTET_STREAM)
+        assertContains(result.body!!, "files <- dir(pattern = \"*.csv\")\n")
+    }
+
+    @Test
+    fun `can POST text to outpack_server`()
+    {
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.TEXT_PLAIN
+        val request = HttpEntity(testPacket, headers)
+        val result = restTemplate.postForEntity(
+                "/outpack/packet/sha256:ad153e5f6720dde3161b229ef73ca2b302fb95a37a092a2c8e3350a8ed6713d4",
+                request, String::class.java
+        )
+        assertSuccess(result)
+        jsonValidator.validateAgainstOutpackSchema(result.body!!, "null-response")
+    }
+
+    @Test
+    @Ignore
```

Comment:
This test ignored pending resolution of the `outpack_server` issue mentioned in the PR descriptiob.